hera_pspec/flags.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+
+mask_generator(nsamples, flags, n_threshold, greedy=False, axis, greedy_threshold, retain_flags=True):
+    """
+    Generates a greedy flags mask from input flags and nsamples arrays
+    
+    Parameters
+    ----------
+    nsamples : numpy.ndarray
+        integer array with number of samples available for each frequency channel at a given LST angle
+    
+    flags : numpy.ndarray
+        binary array with 1 representing flagged, 0 representing unflagged
+        
+    n_threshold : int
+        minimum number of samples needed for a point to remain unflagged
+        
+    greedy : bool
+        greedy flagging is used if true (default is False)
+        
+    axis : int
```

Comment:
This would be better off as a string, and maybe with a name change, e.g. `first='row'`.